### PR TITLE
Gracefully handle null card sortorders in resource report #10949

### LINKED ISSUE #10949: Resource report raises 500 for model with two cards and at least one lacking sortorder
dev/8.0.x:
- create a model with two cards
- publish, make active: at this point, you have two cards lacking a sortorder
- create an instance
- visit resource report

```py
  File "/Users/jwalls/prj/arches/arches/app/views/api.py", line 1258, in get
    permitted_cards = sorted([
                      ^^^^^^^^
TypeError: '<' not supported between instances of 'NoneType' and 'NoneType'
[20/May/2024 08:38:46] "GET /api/resource_report/977a2123-06eb-4751-8912-b5a0c1f4d7f6?v=beta&uncompacted=true HTTP/1.1" 500 113738
```

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
Before, the API resource report would crash on cards with null sortorders.

### Issues Solved

Closes #10949

### Checklist

-   I targeted one of these branches:
    - [x] dev/8.0.x (under development): features, bugfixes not covered below
-   [ ] I added a changelog in arches/releases (no --> unreleased bug)
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [x] Unit tests pass locally with my changes
-   [x] I added tests that prove my fix is effective or that my feature works
-   [x] My test fails on the target branch

#### Ticket Background
*   Sponsored by: Getty Conservation Institut
*   Found by: @jacobtylerwalls
*   Tested by: @ 
*   Designed by: @

### CODE REVIEW

File: arches/app/views/api.py [hunk 1]
```diff
@@ -1555,14 +1555,14 @@ def get(self, request, resourceid):
 
         if "cards" not in exclude:
             readable_nodegroup_ids = [
-                nodegroup.pk
-                for nodegroup in get_nodegroups_by_perm(
+                nodegroup_id
+                for nodegroup_id in get_nodegroups_by_perm(
```
Comment (jacobtylerwalls) [inline arches/app/views/api.py:1559]: This unrelated change (and sim. below) should trickle up from 7.6 soon. Just here so the new unit test that incidentally hits these lines doesn't fail.

**Review by chrabyrd — APPROVED** [review]
lgtm! 👍
